spec/controllers/user_invitations_controller_spec.rb

```diff
@@ -0,0 +1,303 @@
+      end.to raise_error(PermissionError)
+    end
+
+    it "is not allowed for admin users of another company" do
+      expect do
+        signed_in_user :foo_inc_root
+
+        post :create, user: {
+          organization_id: acme.id.to_s,
+          name: "Foo Bar",
+          email: "foobar@email.com",
+          role: "none"
+        }
+      end.to raise_error(PermissionError)
+    end
+
+    it "creates an invite for admin user" do
+      signed_in_user :acme_root
+      expect(SecureRandom).to receive(:hex).and_return("secure_hex")
+
+      post :create, user: {
+        organization_id: acme.id.to_s,
+        name: "Foo Bar",
+        email: "foobar@email.com",
+        role: "none"
+      }
+
+      invitation = UserInvitation.find_by_email("foobar@email.com")
+      expect(invitation).to be
+      expect(invitation.invited_by).to eq(acme_root)
+      expect(invitation.organization).to eq(acme)
+      expect(invitation.name).to eq("Foo Bar")
+      expect(invitation.email).to eq("foobar@email.com")
+      expect(invitation.role).to eq("none")
+      expect(invitation.auth_token).to eq("secure_hex")
+    end
+
+    it "creates an invite for super admin user" do
+      signed_in_user :root
+      expect(SecureRandom).to receive(:hex).and_return("secure_hex")
+
+      post :create, user: {
+        organization_id: acme.id.to_s,
+        name: "Foo Bar",
+        email: "foobar@email.com",
+        role: "admin"
+      }
+
+      invitation = UserInvitation.find_by_email("foobar@email.com")
+      expect(invitation).to be
+      expect(invitation.invited_by).to eq(root)
+      expect(invitation.organization).to eq(acme)
+      expect(invitation.name).to eq("Foo Bar")
+      expect(invitation.email).to eq("foobar@email.com")
+      expect(invitation.role).to eq("admin")
+      expect(invitation.auth_token).to eq("secure_hex")
+    end
+
+    it "normalizes the email address like devise is configured to do" do
+      signed_in_user :acme_root
+      expect(SecureRandom).to receive(:hex).and_return("secure_hex")
+
+      post :create, user: {
+        organization_id: acme.id.to_s,
+        name: "Foo Bar",
+        email: "  FooBar@email.com   \t \n ",
+        role: "admin"
+      }
+
+      invitation = UserInvitation.find_by_email("foobar@email.com")
+      expect(invitation).to be
+      expect(invitation.invited_by).to eq(acme_root)
+      expect(invitation.organization).to eq(acme)
+      expect(invitation.name).to eq("Foo Bar")
+      expect(invitation.email).to eq("foobar@email.com")
+      expect(invitation.role).to eq("admin")
+      expect(invitation.auth_token).to eq("secure_hex")
+    end
+
+    xit "sends an email invite" do
+      signed_in_user :acme_root
+
+      post :create, user: {
+        organization_id: acme.id.to_s,
+        name: "Foo Bar",
+        email: "foobar@email.com",
+        role: "none"
+      }
+
+      # TODO: expect email
+    end
+
+    it "immediately adds the user if they already exist" do
```

Comment:
Add stub of this spec to ensure an email is sent to the applicant to let them know they've been added to a new organization.